src/dmd/mtype.d

```diff
@@ -4169,22 +4169,29 @@ extern (C++) final class TypeFunction : TypeNext
 
     ParameterList parameterList;   // function parameters
 
-    bool isnothrow;             // true: nothrow
-    bool isnogc;                // true: is @nogc
-    bool isproperty;            // can be called without parentheses
-    bool isref;                 // true: returns a reference
-    bool isreturn;              // true: 'this' is returned by ref
-    bool isscope;               // true: 'this' is scope
-    bool isreturninferred;      // true: 'this' is return from inference
-    bool isscopeinferred;       // true: 'this' is scope from inference
-    bool islive;                // is @live
+    private enum FunctionFlag
+    {
+        isnothrow       = 0x0001, // nothrow
+        isnogc          = 0x0002, // is @nogc
+        isproperty      = 0x0004, // can be called without parentheses
+        isref           = 0x0008, // returns a reference
+        isreturn        = 0x0010, // 'this' is returned by ref
+        isscope         = 0x0020, // 'this' is scope
+        isreturninferred= 0x0040, // 'this' is return from inference
+        isscopeinferred = 0x0080, // 'this' is scope from inference
+        islive          = 0x0100, // is @live
+        incomplete      = 0x0200, // return type or default arguments removed
+        inoutParam      = 0x0400, // inout on the parameters
+        inoutQual       = 0x0800, // inout on the qualifier
+    }
+    static assert (FunctionFlag.max <= 0x8000, "flags are stored in a ushort");
```

Comment:
Could also make this more explicit:

```suggestion
    static assert (FunctionFlag.max <= ushort.max, "flags are stored in a ushort");
```

Reply:
No, in this case you cannot have a valid enum member over 0x8000, remember, it's for a set so everything that between 0x8001 and 0xFFFF would cause problems.

Reply:
I know, but then you would also have to check that there are no values between 0x4001 and 0x7FFF, ... (additionally there's the precedent of `STC.safeGroup` and friends).
This assert effectively checks that 1 is equal to 1 ATM.